### PR TITLE
PERF-4050 Allow heuristic Bonsai variants to run auto Genny workloads

### PR DESCRIPTION
**Pull request by mivanova3**
**Jira Ticket:** PERF-4050

**Whats Changed:**  
Allow Genny workload to run with Bonsai Heuristic variant

**Patch testing results:**  
https://spruce.mongodb.com/version/646e04fbe3c331718d4d831a/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Related PRs:**   
DSI: https://github.com/10gen/dsi/pull/1289

### CODE REVIEW

**Review by alyacb — APPROVED** [review]
Thank you Milena! LGTM, mod a request to try re-running [mixed_workloads_genny_rate_limited](https://spruce.mongodb.com/task/sys_perf_linux_intel_1_node_replSet_heuristic_bonsai_mixed_workloads_genny_rate_limited_patch_999ae5bfcdbe49fbde445058c8438833f6a39ec9_646e04fbe3c331718d4d831a_23_05_24_12_44_25?execution=0) on heuristics (and also run it at least once on the sampling variant). The error in the UI looks unrelated, but in case we can't run it successfully, we should file an investigation ticket, leave it commented out, and add a TODO comment mentioning the ticket number.

File: src/workloads/scale/LargeIndexedIns.yml [hunk 1]
```diff
@@ -68,4 +68,6 @@ AutoRun:
       - single-replica-classic-query-engine
       # TODO PERF-4127: Uncomment this build variant after fixing the system failure.
       # - single-replica-sampling-bonsai
+      # TODO PERF-4050:
```
Comment (RuoxinXu) [inline src/workloads/scale/LargeIndexedIns.yml:71]: Should we file a new ticket to unblock this heuristic build variant?

File: src/workloads/scale/MixedWorkloadsGennyRateLimited.yml [hunk 2]
```diff
@@ -192,5 +192,7 @@ AutoRun:
       - replica-noflowcontrol
       - single-replica
       - single-replica-classic-query-engine
+      - single-replica-heuristic-bonsai
+      - single-replica-sampling-bonsai
```
Comment (RuoxinXu) [inline src/workloads/scale/MixedWorkloadsGennyRateLimited.yml:196]: Thanks Milena for adding the sampling variant! Could you please also run a Replica Sampling variant to check that this workload is good to run?

**Review by RuoxinXu — APPROVED** [review]
Thanks Milena! LGTM!

**Review by RuoxinXu — APPROVED** [review]
Thanks Milena! LGTM!

**Review by mivanova3 — COMMENTED** [review]
The evergreen patch is now fine, the failing workloads passed after a restart and the new one passed in both variants.

**Comment by mivanova3** [comment]
evergreen merge

**Pull request merged**